Spec/RestClient.swift

```diff
@@ -1623,13 +1631,24 @@ class RestClient: QuickSpec {
 
                         waitUntil(timeout: testTimeout) { done in
                             do {
-                                try rest.request("get", path: "feature", params: nil, body: nil, headers: nil) { paginatedResult, error in
+                                try rest.request("get", path: "feature", params: nil, body: nil, headers: nil) { paginatedResponse, error in
                                     guard let error = error else {
-                                        fail("Error is nil"); done(); return
+                                        fail("Error is empty"); done(); return
```

Comment:
I don't understand why you expect there to be an error in this case. The semantics of this method is that the request was made, and there was a response, and that's that. There was no error.

The response includes a statusCode that is an "error" result, but the request itself didn't fail. It succeeded, and returned a result.

Also, pls can we have tests for requests that do indeed fail - for example it times out, fails to resolve the host, returns a response body that cannot be parsed, etc.

Reply:
@paddybyers this is making a `GET /feature` which doesn't exist, so the request fails and the error should not be nil. This `guard` simply makes the test fail if the error is nil.

Reply:
No, that's not right. If the server responds to the request, and that response is well-formed (even though it is a 404), then that counts as success *in this API*. There is no `error`, but the response, encapsulated in the `HttpPaginatedResponse`, will indicate that the *content of the response* was an error, appropriately populating `success`, `statusCode`, `errorCode` and `errorMessage`.

RSC19e states that this method does result in an error if the server did not respond, or its response was not well-formed in some way, so it was not possible to return a `HttpPaginatedResponse`.

Reply:
Oh I see now. So the error in nested in the pagination block of the response, not on the first level of the returned object.

Reply:
> No, that's not right. If the server responds to the request, and that response is well-formed (even though it is a 404), then that counts as success *in this API*. There is no `error`, but the response, encapsulated in the `HttpPaginatedResponse`, will indicate that the *content of the response* was an error, appropriately populating `success`, `statusCode`, `errorCode` and `errorMessage`.

That makes it clear, yes. Thanks. I'll fix it.

Reply:
Fixed 861f32b.

Reply:
Fixed 63ebbf5.